chapters/missing-data.qmd

```diff
@@ -0,0 +1,598 @@
+            impute_median_data,
+            impute_linear_data,
+            impute_knn_data,
+            impute_bag_data) %>%
+  mutate(Imputation = factor(Imputation, levels=c("Original Data", "Mean", "Median", "Linear", "k-NN", "Bag")),
+         Observed_Type = paste0(Observed,"_",Type)) %>%
+  mutate(Observed_Type = factor(Observed_Type, levels=c("Group 1_Original", "Group 2_Original", 
+                                                        "Group 1_Imputed", "Group 2_Imputed"))) 
+
+simulation_nonlinear_original_data_small <-
+  simulation_nonlinear_original_data %>%
+  dplyr::select(`Predictor 1`, `Predictor 2`) %>%
+  mutate(Sample = row_number())
+
+imputation_distance_comparison <-
+  bind_cols(
+    simulation_nonlinear_original_data %>% dplyr::select(`Predictor 1`, `Predictor 2`) %>% rename(Orig1 = `Predictor 1`, Orig2 = `Predictor 2`),
+    impute_mean_data %>% dplyr::select(`Predictor 1`, `Predictor 2`) %>% rename(Mean1 = `Predictor 1`, Mean2 = `Predictor 2`),
+    impute_median_data %>% dplyr::select(`Predictor 1`, `Predictor 2`) %>% rename(Median1 = `Predictor 1`, Median2 = `Predictor 2`),
+    impute_linear_data %>% dplyr::select(`Predictor 1`, `Predictor 2`) %>% rename(Linear1 = `Predictor 1`, Linear2 = `Predictor 2`),
+    impute_knn_data %>% dplyr::select(`Predictor 1`, `Predictor 2`) %>% rename(knn1 = `Predictor 1`, knn2 = `Predictor 2`),
+    impute_bag_data %>% dplyr::select(`Predictor 1`, `Predictor 2`) %>% rename(Bag1 = `Predictor 1`, Bag2 = `Predictor 2`)
+  ) %>%
+  mutate(Mean = sqrt((Orig1 - Mean1)^2 + (Orig2 - Mean2)^2),
+         Median = sqrt((Orig1 - Median1)^2 + (Orig2 - Median2)^2),
+         Linear = sqrt((Orig1 - Linear1)^2 + (Orig2 - Linear2)^2),
+         `k-NN` = sqrt((Orig1 - knn1)^2 + (Orig2 - knn2)^2),
+         Bag = sqrt((Orig1 - Bag1)^2 + (Orig2 - Bag2)^2)) %>%
+  dplyr::select(Mean, Median, Linear, `k-NN`, Bag) %>%
+  pivot_longer(
+    everything(),
+    names_to = "Method",
+    values_to = "Distance"
+  ) %>%
+  mutate(Method = factor(Method, levels=c("Mean", "Median", "Linear", "k-NN", "Bag")))
+```
+
+For missing data, imputation is the process of using other existing information (i.e., the predictors) to estimate what each missing value might have been. In other words, we will build an imputation model to fill in the missing column(s) so that we can run the primary machine-learning model. A separate imputation model is required for each column that contains (or could contain) missing data. 
+
+Imputation is a well-researched statistical methodology. This technique has traditionally been used for inferential models, focusing on maintaining the quality of test statistics to support hypothesis testing. As mentioned in earlier chapters, there is an important distinction between the objectives of statistical inference and prediction accuracy. We must consider these competing objectives when deciding how missing values will be imputed.
+
+TODO inference; Lucy's paper; causal inference
```

Comment:
I'd like to contrast prediction versus inference here too. Maybe talk about [using the outcome (or not)](https://journals.sagepub.com/doi/10.1177/09622802241244608) since there is a good connection there.

Reply:
Also, maybe mention that we can avoid bias by weighting by the probability of being missing (or at least give a reference for this).

Reply:
Totally agree--I had meant to include the information about using the outcome but forgot to.  I had not thought about weighting and it makes sense to discuss.

Reply:
@ea70a8e For updates. 

I'm thinking now about skipping the causal inference connection; I'd like to keep this chapter short(ish).